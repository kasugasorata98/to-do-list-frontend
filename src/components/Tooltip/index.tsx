import React, { CSSProperties, ReactNode } from "react";
import { Tooltip as ChakraTooltip } from "@chakra-ui/react";
import { colors } from "@/styles/colors";

const Tooltip: React.FC<{
  style?: CSSProperties;
  children?: ReactNode;
  onClick?: () => void;
  label?: string;
}> = ({ style, children, onClick, label, ...rest }) => {
  return (
    <ChakraTooltip
      style={{
        color: colors.grey,
        ...style,
      }}
      label={label}
      onClick={onClick}
      {...rest}
    >
      {children}
    </ChakraTooltip>
  );
};

export default Tooltip;
