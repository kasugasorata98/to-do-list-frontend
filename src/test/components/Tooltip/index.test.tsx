import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Tooltip from "@/components/Tooltip";

describe("Tooltip component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Tooltip label="This is a tooltip">Hover me</Tooltip>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders children", () => {
    const { getByText } = render(<Tooltip>Tooltip content</Tooltip>);
    expect(getByText("Tooltip content")).toBeInTheDocument();
  });
});
