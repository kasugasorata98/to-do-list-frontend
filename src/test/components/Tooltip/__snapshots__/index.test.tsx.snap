// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component renders correctly 1`] = `
<div>
  <span
    class="css-1baulvz"
    tabindex="0"
  >
    Hover me
  </span>
</div>
`;
